### ISSUE TITLE
Depth buffer creation is unimplemented in OpenGL backend

### ISSUE THREAD

**Issue by caelunshun** [gfx-rs/gfx#2376]
Attempts to create a depth buffer with
```
device.create_image(
    image::Kind::D2(extent.width as Size, extent.height as Size, 1, 1),
    1,
    Format::D32FloatS8Uint,
    image::Tiling::Optimal,
    image::Usage::DEPTH_STENCIL_ATTACHMENT,
    image::StorageFlags::empty(),
);
```
fail with an unimplemented panic. I'm not sure how difficult it is to fix this, but depth buffers seem like a reasonably basic and useful feature.

**Comment by kvark** [comment]
Thank you for the issue! OpenGL is a bit lagging behind in terms of supported functionality, but we aim to bring it on par at some point. Help is welcome! ;)

**Comment by jackmott** [comment]
@caelum100 do you happen to have a small example scene I could test with that you could send in a gist or something?

**Comment by caelum100** [comment]
Here's one from a tutorial: 
https://github.com/Mistodon/gfx-hal-tutorials/blob/master/src/bin/part05-depth.rs

**Comment by caelum100** [comment]
Yes, still an issue.

**Comment by kvark** [comment]
@Thomspoon I don't think this is fixed by #2618. The PR fixes an attempt to use such a texture in a descriptor set, while the issue originally is for the mere fact of creating the texture. Re-opening.

**Comment by Thomspoon** [comment]
This no longer ends in an unimplemented!() call. You can test it with the quad example.

**Comment by ryanw** [comment]
Has this been fixed? When trying to create a depth buffer using `create_image` with `Format::D32FloatS8Uint` I still get `not yet implemented` from this line: https://github.com/gfx-rs/gfx/blob/master/src/backend/gl/src/device.rs#L1156

**Comment by Thomspoon** [comment]
@kvark Two months ago when I was a lowely pupil, I mistakenly said this was fixed when it wasn't, realizing now that this was an issue with the Format type, not the usage flags. However, this can be fixed in #2663 as I got D32Sfloat to work. I'd just need to add the rest of the depth/stencil variants.
